From 2b0859446915a1291dbdb30573d0e485da4e5f6f Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 22 Dec 2021 04:39:52 +0200
Subject: [PATCH] [SW-0] gaudi2: sec: add sec enabled check

Add a security enabled precondition for the sram sf access functions
as locking and unlocking is only required when security is enabled.

Change-Id: I6982c09474c6eb2426bd71cc1148b351fcb0b53f
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 029fa3f..27f69f7 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -828,6 +828,9 @@
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
 	rr_sel.pcie = true;
 
+	if (!preboot_security_is_enabled())
+		return;
+
 	if (en) {
 		/*
 		 * Make SRAM Read-only
@@ -859,6 +862,9 @@
 	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
 	rr_sel.pcie = true;
 
+	if (!preboot_security_is_enabled())
+		return;
+
 	if (en) {
 		/*
 		 * Make SPI Flash Read-only.
